pkg/integration/helpers.go

```diff
@@ -0,0 +1,343 @@
+
+	adminKey, deviceKey string
+
+	adminSrv *server.Server
+	apiSrv   *server.Server
+}
+
+// NewTestSuite creates a Suite for integration tests.
+func NewTestSuite(tb testing.TB, ctx context.Context) *Suite {
+	tb.Helper()
+	cfg, db := config.NewIntegrationTestConfig(ctx, tb)
+
+	// Create or reuse the existing realm
+	realm, err := db.FindRealmByName(realmName)
+	if err != nil {
+		if !database.IsNotFound(err) {
+			tb.Errorf("error when finding the realm %q: %w", realmName, err)
+		}
+		realm = database.NewRealmWithDefaults(realmName)
+		realm.RegionCode = realmRegionCode
+		if err := db.SaveRealm(realm); err != nil {
+			tb.Errorf("failed to create realm %+v: %w: %v", realm, err, realm.ErrorMessages())
+		}
+	}
+
+	// Create new API keys
+	suffix, err := randomString()
+	if err != nil {
+		tb.Errorf("failed to create suffix string for API keys: %w", err)
+	}
+
+	adminKey, err := realm.CreateAuthorizedApp(db, &database.AuthorizedApp{
+		Name:       adminKeyName + suffix,
+		APIKeyType: database.APIUserTypeAdmin,
+	})
+	if err != nil {
+		tb.Errorf("error trying to create a new Admin API Key: %w", err)
+	}
+
+	deviceKey, err := realm.CreateAuthorizedApp(db, &database.AuthorizedApp{
+		Name:       deviceKeyName + suffix,
+		APIKeyType: database.APIUserTypeDevice,
+	})
+	if err != nil {
+		tb.Errorf("error trying to create a new Device API Key: %w", err)
+	}
+
+	return &Suite{
+		cfg:       cfg,
+		db:        db,
+		realm:     realm,
+		adminKey:  adminKey,
+		deviceKey: deviceKey,
+	}
+}
+
+// NewAdminAPIServer runs an Admin API Server and returns a corresponding client.
+func (s *Suite) NewAdminAPIServer(ctx context.Context, tb testing.TB) *AdminClient {
+	srv := s.newAdminAPIServer(ctx, tb)
+	s.adminSrv = srv
+	return NewAdminClient(srv.Addr(), s.adminKey)
+}
+
+// NewAPIServer runs an API Server and returns a corresponding client.
+func (s *Suite) NewAPIServer(ctx context.Context, tb testing.TB) *APIClient {
+	srv := s.newAPIServer(ctx, tb)
+	s.apiSrv = srv
+	return NewAPIClient(srv.Addr(), s.deviceKey)
+}
+
+func (s *Suite) newAdminAPIServer(ctx context.Context, tb testing.TB) *server.Server {
+	// Create the router
+	adminRouter := mux.NewRouter()
+	// Install common security headers
+	adminRouter.Use(middleware.SecureHeaders(ctx, s.cfg.AdminAPISrvConfig.DevMode, "json"))
+
+	// Enable debug headers
+	processDebug := middleware.ProcessDebug(ctx)
+	adminRouter.Use(processDebug)
+
+	// Create the renderer
+	h, err := render.New(ctx, "", s.cfg.APISrvConfig.DevMode)
+	if err != nil {
+		tb.Errorf("failed to create the renderer %v", err)
+	}
+
+	// Setup cacher
+	cacher, err := cache.CacherFor(ctx, &s.cfg.APISrvConfig.Cache, cache.MultiKeyFunc(
+		cache.HMACKeyFunc(sha1.New, s.cfg.APISrvConfig.Cache.HMACKey),
+		cache.PrefixKeyFunc("apiserver:cache:"),
+	))
+	if err != nil {
+		tb.Fatalf("failed to create cacher: %v", err)
+	}
+	tb.Cleanup(func() {
+		if err := cacher.Close(); err != nil {
+			tb.Errorf("got err when cleanup: %v", err)
```

Comment:
```suggestion
			tb.Fatalf("failed to close cacher: %v", err)
```

Reply:
Hey Seth, I just realized I can't resolve suggested edits myself. Could you please resolve this one and also approve the PR? Thanks!